### PR TITLE
Support for custom shell path in `ExecuteAction`

### PR DESCRIPTION
**Pull request by JCSooHwanCho**
### Short description 📝

#3384 introduced custom shellPath to TargetAction. Although ExecutionAction also has this feature, It was missing. So this time, I wanna add this feature.

### Contributor checklist ✅

- [x] The code has been linted using run `./fourier lint tuist --fix`
- [x] The change is tested via unit testing or acceptance testing, or both
- [x] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## Next
+
+### Added
+- Added custom shellPath to ExecuteAction [#5154](https://github.com/tuist/tuist/pull/5154) by [@JCSooHwanCho](https://github.com/JCSooHwanCho)
+
```
Comment (danyf90) [inline CHANGELOG.md:7]: No need to touch the changelog, it's generated from the PR title

File: Sources/ProjectDescription/ExecuteAction.swift [hunk 2]
```diff
@@ -6,9 +6,14 @@ public struct ExecutionAction: Equatable, Codable {
     public let scriptText: String
     public let target: TargetReference?
 
-    public init(title: String = "Run Script", scriptText: String, target: TargetReference? = nil) {
+    /// The path to the shell which shall execute this script.
+    /// if it is nil, "/bin/sh" will be used
```
Comment (danyf90) [inline Sources/ProjectDescription/ExecuteAction.swift:10]: If it is nil it uses the Xcode default, the fact that the default is `/bin/sh` shouldn't belongs here.
Also, it fits on one line.

Same for the similar change in TuistGraph

**Review by danyf90 — CHANGES_REQUESTED** [review]

**Review by kwridan — APPROVED** [review]
Thanks @JCSooHwanCho 

Only suggestion I would make is modifying at least one of the actions in the [`ios_with_custom_scheme` fixture](https://github.com/tuist/tuist/tree/main/projects/tuist/fixtures/ios_app_with_custom_scheme) to help demo / showcase this new feature 🙂

Thanks!

**Review by danyf90 — APPROVED** [review]

**Comment by JCSooHwanCho** [comment]
@kwridan Thank you! I applied.

**Comment by danyf90** [comment]
@allcontributors add @JCSooHwanCho  for code

**Comment by JCSooHwanCho** [comment]
@danyf90 Is there any problem in unit test process? It seems like no test failure, but the process failed with error code. can I get any clue about it?

**Comment by danyf90** [comment]
> @danyf90 Is there any problem in unit test process? It seems like no test failure, but the process failed with error code. can I get any clue about it?

This one is failing:     ✖ testXcodeProjStructureDoesNotChangeAfterRegeneration, XCTAssertEqual failed: ("9") is not equal to ("0")

Looks like an issue caused by the bump in XcodeProj 🤔 

@kwridan do you have any experience with it?

**Comment by JCSooHwanCho** [comment]
@danyf90 @kwridan It turns out that this test fail is result from [this change](https://github.com/tuist/XcodeProj/pull/739)

```
final public class XcodeProj : Equatable {
   // ...
    public var userData: [XcodeProj.XCUserData] <- fail point
   // ... 
}

final public class XCUserData : Equatable, XcodeProj.Writable {
    // ...
    public var schemeManagement: XcodeProj.XCSchemeManagement?  <- fail point
    // ...
}

public struct XCSchemeManagement : Codable, Equatable, XcodeProj.Writable {
   public var schemeUserState: [XcodeProj.XCSchemeManagement.UserStateScheme]? <- fail point
}
```

schemeUserState order is random and equality check is failing

**Comment by JCSooHwanCho** [comment]
https://opensource.apple.com/source/CommonCrypto/CommonCrypto-60027/CommonCrypto.xcodeproj/xcuserdata/stripes.xcuserdatad/xcschemes/xcschememanagement.plist.auto.html

according to this sample data, It seems that schemeUserState should be dictionary.

**Comment by kwridan** [comment]
Thanks for investigating this @JCSooHwanCho 🙏

Ouch looks like reads are unstable due to the Dictionary > Array conversion, looks like we missed this when the feature was first introduced even though stability tests were added 😅

https://github.com/tuist/XcodeProj/pull/565/files#diff-55065e34288606fdfc846acb7583090de95cd05838908667a842847f9cd33693R7

Will take a look to see if we can do a sort internally to address this.


> https://opensource.apple.com/source/CommonCrypto/CommonCrypto-60027/CommonCrypto.xcodeproj/xcuserdata/stripes.xcuserdatad/xcschemes/xcschememanagement.plist.auto.html
> 
> according to this sample data, It seems that schemeUserState should be dictionary.

Indeed it appears there's a discrepancy, I wonder if the intention was to preserve the order of the items as they appeared in the plist - but the fact there is an instability leads me to believe that is not the case

**Comment by kwridan** [comment]
Posted https://github.com/tuist/XcodeProj/pull/758

**Comment by danyf90** [comment]
@JCSooHwanCho could you please try using `XcodeProj 8.10.0`? 🙏

**Comment by JCSooHwanCho** [comment]
@danyf90 Done!

**Comment by JCSooHwanCho** [comment]
@danyf90 Another faiiure appeared. have you ever experienced with it? It's from plugin acceptance test.

**Pull request merged**